nextstrain/cli/pathogens.py

```diff
@@ -845,62 +856,54 @@ def read_pathogen_registration(path: Path) -> Dict:
     return registration
 
 
-# XXX TODO SOON: This logic should probably move into nextstrain.org endpoints
-# (and start us down the road of a real Nextstrain pathogen registry).  That
-# would also give us insight into usage and allows us the flexibility to move
-# away from Git-based distribution in the future.
-#   -trs, 3 Feb 2025
-def github_repo_latest_ref(repo_name: str) -> Optional[str]:
+# We query a nextstrain.org API instead of querying GitHub's API directly for a
+# few reasons.
+#
+#  1. It gives us greater flexibility to move away from Git/GitHub-based
+#     versioning and distribution in the future.
+#  2. It gives us insight into usage.
+#  3. It starts us down the road of a real Nextstrain pathogen registry.
+#
+#   -trs, 22 April 2025
+def nextstrain_repo_latest_version(name: str) -> Optional[str]:
     """
-    Queries a GitHub *repo_name* to return the name of the highest version tag,
-    if any, otherwise the name of the default branch.
+    Queries a Nextstrain pathogen repo *name* to return the latest version,
+    i.e. the name of the highest version tag, if any, otherwise the name of the
+    default branch.
     """
+    org = URL(NEXTSTRAIN_DOT_ORG)
+
+    versions_url = URL(f"https://nextstrain.org/pathogen-repos/{urlquote(name)}/versions")._replace(scheme = org.scheme, netloc = org.netloc)
```

Comment:
I'm confused by the split of `org` and `versions_url` here...I thought  25e6560a3e30780a8f6012149a25a678c83b54f9 would allow adding the base to the URL:

```suggestion
versions_url = URL(f"pathogen-repos/{urlquote(name)}/versions", NEXTSTRAIN_DOT_ORG)
```

Reply:
Ah, good question! There's indeed a subtly here.

We want to take only the scheme and netloc (host:port) from `NEXTSTRAIN_DOT_ORG`, to be consistent with how the override is used by `nextstrain remote` and `nextstrain login` commands, e.g.

https://github.com/nextstrain/cli/blob/8007c4790c55ebb17f452956c4e530d479bd2dff/nextstrain/cli/remote/__init__.py#L123-L125

Using it as a base URL, as in `URL(…, NEXTSTRAIN_DOT_ORG)`, would also allow potential path segments to be taken/used as well, e.g.

```python-console
>>> str(URL(f"a/b/c", "https://example.com/x/y/z/"))
'https://example.com/x/y/z/a/b/c'
```

Reply:
Ah, I see. Thanks for the explanation!